### ISSUE TITLE
insert cookie and got another error

### ISSUE THREAD

**Issue by tszjqgs** [linkedtales/scrapedin-linkedin-crawler#24]
This time I filled into the cookie field and were able to start scrap, but after only one profile scraping, it throws an error :

scrapedin: 2019-10-18T20:09:22.541Z info: [profile] finished scraping url: https://www.linkedin.com/in/linda-b32a4815a
(node:9720) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
    at crawl (/home//Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:9720) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:9720) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
scrapedin: 2019-10-18T20:09:23.286Z info: [profile] finished scraping url: https://www.linkedin.com/in/tiffanie-walker-26512757
(node:9720) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
    at crawl (/home/Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:9720) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)

-----------so , any signs of issues ? thanks

**Comment by tszjqgs** [comment]
could you please provide a sample usage code with cookie involved ?

**Comment by leonardiwagner** [comment]
Hi @tszjqgs , really sorry, the logger should be inside the `try/catch` block as well, please update via `git pull` or update your code manually with this change: https://github.com/linkedtales/scrapedin-linkedin-crawler/pull/25/files

**Comment by tszjqgs** [comment]
Hi Leo,
is is possible to give an example of cookie usage ?

I export the cookie and copy the part of it look like below to the config.json
after cookieFile field . and running error message""

{
    "domain": ".www.linkedin.com",
    "expirationDate": 160353.006772,
    "hostOnly": false,
    "httpOnly": true,
    "name": "li_at",
    "path": "/",
    "sameSite": "unspecified",
    "secure": true,
    "session": false,
    "storeId": "0",
    "value": "",
    "id": 23
}


error msg:

{
    "domain": ".www.linkedin.com",
    "expirationDate": 13.006772,
    "hostOnly": false,
    "httpOnly": true,
    "name": "li_at",
    "path": "/",
    "sameSite": "unspecified",
    "secure": true,
    "session": false,
    "storeId": "0",
    "value": "SZIR5kmh5V8",
    "id": 23
}

**Comment by leonardiwagner** [comment]
@tszjqgs  you have to create a `cookies.json` file to place those cookies info, your files should be like this:
```
-src/
  index.js
-config.json
-cookies.json <- you should create this file
-package.json
```

then, in `config.json` you should inform that file:
```
"cookiesFile": "./cookies.json"
```

Notice the content of `cookies.json` file is an array of objects, should be like that:
```
[
  {
    "domain": ".linkedin.com",
    "expirationDate":...
  },
 {
..}
]
```

my cookies file generally has 250~300 lines

**Comment by tszjqgs** [comment]
thanks , it works now for cookies .
but when start crawling , there are no crawledprofiles folder created and no result shows up 
and I do can be able to see the content of the profile link in browser

**Comment by leonardiwagner** [comment]
hf @tszjqgs ,you could set a fixed directory on `config.json`, example:

`"saveDirectory":"home/Documents/crawledProfiles"`
or in windows `"saveDirectory": "C:/crawledProfiles"`

**Comment by tszjqgs** [comment]
right now and before , it has been "saveDirectory": "./crawledProfiles",  which will show at the same path with where config.json is . but the main point is at this time , the crawledProfiles folder is not created and no information was captured and the error is like this ,I did not happen before .


2019-10-23T22:46:02.717Z info: starting scraping: https://www.linkedin.com/in/marquita-mcgee-181900100
scrapedin: 2019-10-23T22:46:02.717Z info: [profile] starting scraping url: https://www.linkedin.com/in/marquita-mcgee-181900100
scrapedin: 2019-10-23T22:46:07.485Z warn: [profile] profile selector was not found
2019-10-23T22:46:07.486Z error: error on crawling profile: https://www.linkedin.com/in/dolores-bailey-0a7b6b64?trk=pub-pbmap 
 Error: linkedin: profile not found
(node:27156) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
    at crawl (/home/Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
    at <anonymous>
(node:27156) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:27156) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
scrapedin: 2019-10-23T22:46:07.915Z warn: [profile] profile selector was not found
2019-10-23T22:46:07.916Z error: error on crawling profile: https://www.linkedin.com/in/marquita-mcgee-181900100 
 Error: linkedin: profile not found
(node:27156) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
    at crawl (/home/Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
    at <anonymous>
(node:27156) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)

**Comment by tszjqgs** [comment]
right now and before , it has been "saveDirectory": "./crawledProfiles", which will show at the same path with where config.json is . but the main point is at this time , the crawledProfiles folder is not created and no information was captured and the error is like this ,I did not happen before .
2019-10-23T22:46:02.717Z info: starting scraping: https://www.linkedin.com/in/marquita-mcgee-181900100
scrapedin: 2019-10-23T22:46:02.717Z info: [profile] starting scraping url: https://www.linkedin.com/in/marquita-mcgee-181900100
scrapedin: 2019-10-23T22:46:07.485Z warn: [profile] profile selector was not found
2019-10-23T22:46:07.486Z error: error on crawling profile: https://www.linkedin.com/in/dolores-bailey-0a7b6b64?trk=pub-pbmap
Error: linkedin: profile not found
(node:27156) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
at crawl (/home/Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
at
(node:27156) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:27156) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
scrapedin: 2019-10-23T22:46:07.915Z warn: [profile] profile selector was not found
2019-10-23T22:46:07.916Z error: error on crawling profile: https://www.linkedin.com/in/marquita-mcgee-181900100
Error: linkedin: profile not found
(node:27156) UnhandledPromiseRejectionWarning: ReferenceError: relatedProfiles is not defined
at crawl (/home/Downloads/scrapedin-linkedin-crawler/src/crawler.js:31:55)
at
(node:27156) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)

------------------------------------------------------------------
发件人：Wagner Leonardi <notifications@github.com>
发送时间：2019年10月24日(星期四) 05:04
收件人：linkedtales/scrapedin-linkedin-crawler <scrapedin-linkedin-crawler@noreply.github.com>
抄　送：tszjqgs <jason4ever8@aliyun.com>; Mention <mention@noreply.github.com>
主　题：Re: [linkedtales/scrapedin-linkedin-crawler] insert cookie and got another error (#24)

hf @tszjqgs ,you could set a fixed directory on config.json, example:
"saveDirectory":"home/Documents/crawledProfiles"
 or in windows "saveDirectory": "C:/crawledProfiles"

**Comment by leonardiwagner** [comment]
@tszjqgs are you using Windows? I can't reproduce that :cry:

**Comment by tszjqgs** [comment]
{ profileLegacy: undefined,
profileAlternative:
 { name: 'LaKeisha Anderson',
   headline: 'School bus driver/ culinary arts (baking and pastry) school',
   location: 'Chicago, Illinois',
   connections: '35',
   summary: 'Experienced School Bus Driver skilled in Catering, Food & Beverage, Event Planning, Microsoft Excel, and Customer Service. Strong operations professional.' },


aboutLegacy: undefined,
aboutAlternative:
 { text: 'Experienced School Bus Driver skilled in Catering, Food & Beverage, Event Planning, Microsoft Excel, and Customer Service. Strong operations professional.' },
positions: [],
educations:
 [ { title: 'kennedy king',
     degree: 'culinary arts',
     date1: '2003',
     date2: '2006' },
   { title: 'dunbar vocational career academy',
     date1: '1993',
     date2: '1997' } ],
skills:
 [ { title: 'Cooking', count: '2' },
   { title: 'Nutrition', count: '2' },
   { title: 'Customer Service', count: '1' },
   { title: 'Social Media' },
   { title: 'Event Planning' },
   { title: 'Sales' },
   { title: 'Event Management' },
   { title: 'Catering' },
   { title: 'Food' },
   { title: 'Microsoft Office' },
   { title: 'Microsoft Word' },
   { title: 'Microsoft Excel' },
   { title: 'Team Building' },
   { title: 'Coaching' },
   { title: 'Hospitality' },
   { title: 'Social Networking' } ],
recommendations: { givenCount: '0', receivedCount: '0', given: [], received: [] },
accomplishments: [],
peopleAlsoViewed:
 [ { user: 'https://www.linkedin.com/in/davieschr/',
     text: 'Account Manager at C.H. Robinson' } ],
volunteerExperience: [],
profile:
 { name: 'LaKeisha Anderson',
   headline: 'School bus driver/ culinary arts (baking and pastry) school',
   location: 'Chicago, Illinois',
   connections: '35',
   summary: 'Experienced School Bus Driver skilled in Catering, Food & Beverage, Event Planning, Microsoft Excel, and Customer Service. Strong operations professional.' } }

https://www.linkedin.com/in/lakeisha-anderson-96527a1b/


------------------------------------------------------------------
发件人：Wagner Leonardi <notifications@github.com>
发送时间：2019年11月3日(星期日) 08:08
收件人：linkedtales/scrapedin-linkedin-crawler <scrapedin-linkedin-crawler@noreply.github.com>
抄　送：tszjqgs <jason4ever8@aliyun.com>; Mention <mention@noreply.github.com>
主　题：Re: [linkedtales/scrapedin-linkedin-crawler] insert cookie and got another error (#24)

Closed #24.

**Comment by tszjqgs** [comment]
I use ubuntu
------------------------------------------------------------------
发件人：Wagner Leonardi <notifications@github.com>
发送时间：2019年11月3日(星期日) 08:08
收件人：linkedtales/scrapedin-linkedin-crawler <scrapedin-linkedin-crawler@noreply.github.com>
抄　送：tszjqgs <jason4ever8@aliyun.com>; Mention <mention@noreply.github.com>
主　题：Re: [linkedtales/scrapedin-linkedin-crawler] insert cookie and got another error (#24)

@tszjqgs are you using Windows? I can't reproduce that 😢
